### PR TITLE
♻️ Cleanup AnimationPlayer's private state

### PR DESCRIPTION
**Pull request by jridgewell**
Part of #14761.

### CODE REVIEW

File: src/animation.js [hunk 1]
```diff
@@ -197,6 +196,14 @@ class AnimationPlayer {
     this.task_ = this.vsync_.createAnimTask(this.contextNode_, {
       mutate: this.stepMutate_.bind(this),
     });
+
+
```
Comment (rsimha) [inline src/animation.js]: nit: Remove one (or both) blank line(s).

File: src/animation.js [hunk 2]
```diff
@@ -197,6 +196,14 @@ class AnimationPlayer {
...
+    if (this.vsync_.canAnimate(this.contextNode_)) {
+      this.task_(this.state_);
+    } else {
+      dev().warn(TAG_, 'cannot animate');
+      this.complete_(/* success */ false, /* dir */ 0);
```
Comment (rsimha) [inline src/animation.js]: Curious: isn't `complete_` also private state that should be removed?

File: src/animation.js [hunk 3]
```diff
@@ -197,6 +196,14 @@ class AnimationPlayer {
...
+    }
```
[the remark this answers is not in the document]
Comment (alabiaga) [reply src/animation.js]: +1. right. I am not sure if this is desired.

File: src/animation.js (hunk 2 above)
Comment (jridgewell) [reply src/animation.js]: I don't understand your question? It's a private method called from multiple locations.

I'm only interested in removing private fields that are uncalled, or called from external locations (outside the class scope, meaning the field isn't private at all).

File: src/animation.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (jridgewell) [reply src/animation.js]: The only call to `start_` (notice private) is immediately after the constructor. I can make it a public method, but then why construct then call start?

File: src/animation.js (hunk 1 above)
Comment (jridgewell) [reply src/animation.js]: Done.

**Comment by jridgewell** [comment]
> Will let other reviewers approve, since I'm not too familiar with this code.

Dima's owns the code, and he's busy. Falls on you 😉.

File: src/animation.js (hunk 2 above)
Comment (rsimha) [reply src/animation.js]: Makes sense. Carry on.

File: src/animation.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (rsimha) [reply src/animation.js]: Also makes sense.

[the remark this answers is not in the document]
Comment (alabiaga) [reply src/animation.js]: seems like an odd API though. But then again I see that AnimationPlayer is not exported and used anywhere else but in here.

**Pull request merged**